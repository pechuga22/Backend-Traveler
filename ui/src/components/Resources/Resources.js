import React from 'react'
import { Image, Embed } from 'semantic-ui-react'

import './Resources.less'

export default function Resources({ resources }) {
  const typeResource = resources.map((resource) => {
    const { type, source } = resource
    if (type === 'image') {
      return (
        <Image
          src={source}
          size='large'
          className='img resource'
          rounded
          centered
        />
      )
    }
    if (type === 'video') {
      return <Embed url={source} />
    }
    return null
  })
  return <div>{typeResource}</div>
}
