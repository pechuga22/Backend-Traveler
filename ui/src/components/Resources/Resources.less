

.ui.large.image.img.resource {
  height: 200px;
  object-fit: cover;
}

.ui.embed {
  margin: 0.7rem auto;
  border-radius: 6px;
}
