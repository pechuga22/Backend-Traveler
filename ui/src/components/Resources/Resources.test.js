import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Resources from './Resources'

const resources = [
  {
    type: 'image',
    source:
      'https://cdn.zeplin.io/5e1f6633eaa26ca7c49cab85/assets/B1D1D834-4632-4645-9F3E-3817DDBEDB34.png',
  },
  {
    type: 'video',
    source: 'https://www.youtube.com/watch?v=E_WQOBXE7us',
  },
]

describe('./components/Resources/Resources.js', () => {
  describe('.render()', () => {
    it('should render the Resources component', () => {
      const wrapper = shallow(<Resources resources={resources} />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('should have class into Experiences component', () => {
      const component = shallow(<Resources resources={resources} />)
      expect(component.find('.resource').length).toEqual(1)
    })
  })
})
