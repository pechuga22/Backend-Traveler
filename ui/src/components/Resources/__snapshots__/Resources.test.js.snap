// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/Resources/Resources.js .render() should render the Resources component 1`] = `
<div>
  <Image
    as="img"
    centered={true}
    className="img resource"
    rounded={true}
    size="large"
    src="https://cdn.zeplin.io/5e1f6633eaa26ca7c49cab85/assets/B1D1D834-4632-4645-9F3E-3817DDBEDB34.png"
    ui={true}
  />
  <Embed
    url="https://www.youtube.com/watch?v=E_WQOBXE7us"
  />
</div>
`;
